File: HakerRank/algorithms/strings/reduced_string.cs
/*
author: Daniel Lozano
source: HAckerRank(https://www.hackerrank.com )
problem name: Algorithms > String > Super Reduced String
problem url: https://www.hackerrank.com/challenges/reduced-string/problem
*/

using System;

namespace HackerRank
{
    class Program
    {
        static void Main()
        {
            string line = Console.ReadLine();

            int count = 0;
            while (count < line.Length - 1)
            {
                if (line[count] == line[count + 1])
                {
                    line = line.Remove(count, 2);
                    count = -1;
                }
                count++;
            }

            if (String.IsNullOrEmpty(line))
                Console.Write("Empty String");
            else
                Console.Write(line);
        }
    }
}

File: HakerRank/algorithms/sorting/reduced_string.py
line = input()

def remove_char(str, n):
    first_part = str[:n]
    last_part = str[n + 1:]
    return first_part + last_part

count = 0
while count < len(line) - 1:
    if line[count] == line[count + 1]:
        line = line[:count] + line[count + 2:]
        count = -1
    count += 1

if line is not str():
    print(line)
else:
    print("Empty String")